src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyout.cs

```diff
@@ -40,9 +40,24 @@ private void OnItemsVectorChanged(IObservableVector<MenuFlyoutItemBase> sender,
 					default:
 						break;
 				}
+
+				SetFlyoutItemsDataContext();
 			}
 		}
 
+		internal protected override void OnDataContextChanged(DependencyPropertyChangedEventArgs e)
+		{
+			base.OnDataContextChanged(e);
+
+			SetFlyoutItemsDataContext();
+		}
+
+		private void SetFlyoutItemsDataContext()
+		{
+			// This is present to force the dataContext to be passed to the popup of the flyout since it is not directly a child in the visual tree of the flyout. 
+			Items?.ForEach(item => item?.SetValue(MenuFlyoutItem.DataContextProperty, this.DataContext, precedence: DependencyPropertyValuePrecedences.Inheritance));
```

Comment:
For readability, you should split long loc like this into multiple lines:
```
Items?.ForEach(item => item?.SetValue(
    MenuFlyoutItem.DataContextProperty,
    this.DataContext, 
    precedence: DependencyPropertyValuePrecedences.Inheritance
));
```

Reply:
will do